From b0eee4c8d7fe70f991698b8ee5ad9f883fc65b20 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Sun, 28 Apr 2019 11:52:00 +0300
Subject: [PATCH] [SW-0] pll: move sys ref clk definition to header

Currently, all habanalabs devices use the same SoC reference clock.
Although, this might seem like a "not so good" design, move the
frequency define to the common header.
This way we can use it for calculations in generic files.
As an example, switch bypass cycles calculation to use it.

Change-Id: I278fd3e082db36fcac1731be386bee8a0ccb94b6
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/common/pll.h b/common/pll.h
index 15c74c7..0f65ea3 100644
--- a/common/pll.h
+++ b/common/pll.h
@@ -25,9 +25,10 @@
 #define HL_PLL_MAX_WAIT_TIME	20	/* 20 us */
 #define HL_PLL_LOCK_TOUT	5000	/* 5 ms */
 
+#define SYS_REF_CLOCK		50000000ULL
 #define USEC_PER_SEC		1000000
 #define CPU_CYCLES_PER_USEC	(HL_CPU_CYCLES_PER_USEC / USEC_PER_SEC)
-#define BYPASS_CYCLES_PER_USEC	(50000000 / USEC_PER_SEC)
+#define BYPASS_CYCLES_PER_USEC	(SYS_REF_CLOCK / USEC_PER_SEC)
 #define MIN_CYCLES_PER_ITER	5
 
 #define mmPSOC_MME_PLL_BASE	0xC71000
diff --git a/goya/goya.c b/goya/goya.c
index 4365e42..a61e384 100644
--- a/goya/goya.c
+++ b/goya/goya.c
@@ -16,7 +16,6 @@
 #include "pll.h"
 
 #define BTL_COUNTER_MAX		BIT(22)
-#define SYS_REF_CLOCK		50000000
 
 static int preboot_wait_for_bit(unsigned long reg, unsigned long timeout_ns)
 {
